articles/libraries/auth0js/index.md

```diff
@@ -39,591 +39,178 @@ Install via [bower](http://bower.io):
-
-```js
-var auth0 = new Auth0({
-  domain:       'mine.auth0.com',
-  clientID:     'dsa7d77dsa7d7',
-  callbackURL:  'http://my-app.com/callback'
-  // responseType not set (defaults to 'code')
+  audience: 'url:auth:some-audience',
+  scope: 'read:something write:otherthing',
+  responseType: 'token',
+  redirectUri: 'https://example.com/auth/callback'
 });
 ```
 
-On successful login, Auth0 will redirect to your `callbackURL` with an appended authorization `code` query parameter. Unlike the SPA scenario, this `code` value should get processed completely server-side.
+### Custom Username and Password
 
-> Note: Server-side processing of the `code` looks something like this: Using whichever [Auth0 server-side SDK](https://auth0.com/quickstart/webapp) necessary, the endpoint on the server should exchange the `code` for an `access_token` and `id_token` and optionally a full user profile.  It should then set some kind of local session cookie, which is what enables a user to be "logged in" to the website and usually contains data from the user profile.  It should finally redirect the user back to a meaningful page.
-
-#### Popup Mode
-
-Besides Redirect Mode, the `login` method also supports Popup Mode, which you enable by passing `popup: true` in the `options` argument. In this mode the browser will *not* be redirected to a separate login page.  Instead Auth0 will display a popup window where the user enters their credentials. The advantage of this approach is that the original page (and all of its state) remains intact, which can be important, especially for certain Single Page Apps.
-
-> **WARNING**: While Popup Mode does have the advantage of preserving page state, it has some issues. Often times users have popup blockers that prevent the login page from even displaying. There are also known issues with mobile browsers. For example, in recent versions of Chrome on iOS, the login popup does not get closed properly after login (see an example [here](https://github.com/auth0/lock/issues/71)). For these reasons, we encourage developers to favor Redirect Mode over Popup Mode, even with Single Page Apps.
-
-In Popup Mode you also have no need to be redirected back to the application, since, once the user has logged in, the popup is simply closed. Instead Auth0 uses the `login` method's `callback` argument to return control to your client-side application, for both failed and successful logins. Along with the `err` argument, `callback` should also receive a `result` argument with the following properties: `idTokenPayload, idToken, accessToken, state` (and optionally `refreshToken` if the `offline_access` scope has been requested):
+The `client.login` method can be used for username and password (database) logins that require a custom form, as opposed to the Auth0 [hosted login page](/hosted-pages/login).
```

Comment:
Maybe:

client.login can be used for when you need to request username and password and authenticate from your site instead of using the hosted login page